File: GFG Solutions/Odd Even Problem/oddEven.py
from collections import Counter


class Solution:
    def oddEven(self, s: str) -> str:
        freq = Counter(s)

        x = 0
        y = 0

        for char, count in freq.items():
            # Get 1-based position in the alphabet
            position = ord(char) - ord('a') + 1

            if position % 2 == 0:  # Even position in alphabet
                if count % 2 == 0:  # Even frequency
                    x += 1
            else:  # Odd position in alphabet
                if count % 2 == 1:  # Odd frequency
                    y += 1

        # Step 4: Calculate the sum and determine if it's even or odd
        if (x + y) % 2 == 0:
            return "EVEN"
        else:
            return "ODD"

File: GFG Solutions/Odd Even Problem/oddEven.cpp
#include <unordered_map>
#include <string>
using namespace std;
class Solution
{
public:
    string oddEven(string s)
    {
        unordered_map<char, int> freq;

        // Step 1: Calculate the frequency of each character
        for (char c : s)
        {
            freq[c]++;
        }

        int x = 0;
        int y = 0;

        // Step 2: Determine the positions and update x and y accordingly
        for (const auto &pair : freq)
        {
            char charKey = pair.first;
            int count = pair.second;
            int position = charKey - 'a' + 1; // Get 1-based position in the alphabet

            if (position % 2 == 0)
            { // Even position in alphabet
                if (count % 2 == 0)
                { // Even frequency
                    x++;
                }
            }
            else
            { // Odd position in alphabet
                if (count % 2 == 1)
                { // Odd frequency
                    y++;
                }
            }
        }

        // Step 3: Calculate the sum and determine if it's even or odd
        if ((x + y) % 2 == 0)
        {
            return "EVEN";
        }
        else
        {
            return "ODD";
        }
    }
};